iree/compiler/Conversion/Common/LinalgBufferizePass.cpp

```diff
@@ -694,6 +696,18 @@ void LinalgBufferizePass::runOnFunction() {
     return signalPassFailure();
   }
 
+  // Forward buffers assigned to tensor.cast result to its input.
+  if (funcOp
+          .walk([&](tensor::CastOp op) -> WalkResult {
+            auto outBuffer = bvm.lookup(op.dest());
+            if (!outBuffer) return failure();
```

Comment:
mmmm thinking about that I think we shouldn't fail here (e.g case where `tensor.cast` is between two linalg ops)
so we should just skip and case in conversionDispatch ?

Reply:
I dont follow what you mean by `conversionDispatch`, so not able to follow the comment.

1) No need to fail here. If the output has no buffer assigned it should just create a memref allocation for correctness.
2) I find walking this forward is a bit off, since it is really propogating information backwards. The loop below is doing this in an atrocious way. We could just merge the two?

Reply:
> I dont follow what you mean by `conversionDispatch`, so not able to follow the comment.
> 
`conversionDispatch` is the last step walking froward and mapping input buffers to output tensors.

> 1. No need to fail here. If the output has no buffer assigned it should just create a memref allocation for correctness.
> 2. I find walking this forward is a bit off, since it is really propogating information backwards. The loop below is doing this in an atrocious way. We could just merge the two?



Good point walking backward here since this is  a result_buffer -> input_tensors mapping.
Regarding allocating buffers for the output `tensor.cast`  doesn't do any data movements so the buffers should be forward (the underlying buffer type should match by construction) and we should never need to alloc a buffer for a `tensor.cast` except for cases where a cast will never get stored which will endup DCE anyway. 

Updated the code so its now handled as the following:

1- Walk backward and forward output buffers (if mapped) to input tensors 
2- Walk forward and forward input buffers to output tensors.

Reply:
I looked at the updated code it looks fine, but a point about `tensor.cast`. The fact that it is a tensor operation already implies it has "copy" semantics (unlike `memref.cast` which has aliasing semantics). So lowering this to buffer already means that we are introducing aliases. In most cases its fine, but there can be subtle SSA use-def patterns that might end up forcing a copy here. As written right now, this will just crash in `bvm.lookup`. Thats fine for now, but if we do hit a crash there, we know why.